recirq/quantum_chess/quantum_board.py

```diff
@@ -110,8 +101,24 @@ def perform_moves(self, *moves) -> bool:
         return did_it_move
 
     def undo_last_move(self) -> bool:
-        """Undo is not yet implemented."""
-        return False
+        """Undoes the last move.
+
+        Instead of undoing the last move, this function resets
+        the board to the initial state and replays the move history
+
+        """
+        self.accumulations_valid = False
+        
+        # Store current move history...
+        current_move_history = self.move_history.copy()
```

Comment:
Don't we need to change self.move_history?  Do we need to copy it?
Why can't we just modify in place (`delete self.move_history[-1]` or something similar).  
I don't think this works if we undo twice.

Reply:
Calling with_state clears the move history. We need to copy it to replay the moves from the initial basis state.